spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -181,8 +181,11 @@ Pair<int[], Integer> buildPosDelRowIdMapping(PositionDeleteIndex deletedRowPosit
         if (!deletedRowPositions.isDeleted(originalRowId + rowStartPosInBatch)) {
           posDelRowIdMapping[currentRowId] = originalRowId;
           currentRowId++;
-        } else if (hasIsDeletedColumn) {
-          isDeleted[originalRowId] = true;
+        } else {
+          if (hasIsDeletedColumn) {
+            isDeleted[originalRowId] = true;
+          }
+          deletes.incrementDeleteCount();
```

Comment:
Are we sure that this `else` block is refactored correctly?

Is it possible for `hasIsDeletedColumn` to be false but we still need to increment `deletes.incrementDeleteCount()` (at least in the context of position deletes)?

Reply:
Here is our discussion on this, https://github.com/apache/iceberg/pull/4588#discussion_r945959579. In general, I don't have strong option on either side, but prefer think this metric is orthogonal to metadata column `_deleted`.

Reply:
Oh I see. Yeah then we’d want to keep the metric whether we had the column or not.

Thanks for pointing me in the right direction.

Reply:
Yes, this is correct.
The structure of the logic is
```
        if (...) {
          // row is not deleted
        } else {
          // row is deleted
          if (hasIsDeletedColumn) { ... }
          deletes.incrementDeleteCount();
        }
```
In the else clause, we increment the counter regardless of `hasIsDeletedColumn` because this is the case where the row is deleted.

Reply:
One small nit: Can we add a blank line between the if block and the metrics increment? It’s in-line with the style guide and I think would reduce the tendency to wonder if the final else is intentional.

Reply:
Sure. I have added blank lines after non-final if blocks.